inc/impact.class.php

```diff
@@ -0,0 +1,1091 @@
+         $total = count($DB->request([
+            'FROM'   => ImpactRelation::getTable(),
+            'WHERE'  => [
+               'OR' => [
+                  [
+                     'itemtype_source' => get_class($item),
+                     'items_id_source' => $item->getID(),
+                  ],
+                  [
+                     'itemtype_impacted' => get_class($item),
+                     'items_id_impacted' => $item->getID(),
+                  ]
+               ]
+            ]
+         ]));
+      } else if (is_a($item, "CommonITILObject")) {
+         // Tab name for an ITIL object : always 0
+         $total = 0;
+      } else {
+         // Neither an asset nor an CommonITILObject
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) is not a valid target for impact analysis."
+         );
+      }
+
+      return self::createTabEntry(
+         _n('Impact ', 'Impacts ', Session::getPluralNumber()),
+         $total
+      );
+   }
+
+   public static function displayTabContentForItem(
+      CommonGLPI $item,
+      $tabnum = 1,
+      $withtemplate = 0
+   ) {
+      global $CFG_GLPI;
+
+      $class = get_class($item);
+
+      // Only enabled for CommonDBTM
+      if (!is_a($item, "CommonDBTM")) {
+         throw new InvalidArgumentException(
+            "Argument \$item ($class) must be a CommonDBTM)."
+         );
+      }
+
+      $ID = $item->getID();
+
+      // Don't show the impact analysis on new object
+      if ($item->isNewID($ID)) {
+         return false;
+      }
+
+      // Check READ rights
+      $itemtype = $item->getType();
+      if (!$itemtype::canView()) {
+         return false;
+      }
+
+      // For an ITIL object, load the first linked element by default
+      if (is_a($item, "CommonITILObject")) {
+         $linked_items = $item->getLinkedItems();
+
+         // Search for a valid linked item of this ITILObject
+         $found = false;
+         foreach ($linked_items as $linked_item) {
+            $class = $linked_item['itemtype'];
+            if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+               self::printAssetSelectionForm($linked_items);
+               $found = true;
+               $item = new $class;
+               $item->getFromDB($linked_item['items_id']);
+               break;
+            }
+         }
+
+         // No valid linked item were found, tab shouldn't be visible
+         if (!$found) {
+            return true;
+         }
+      }
+
+      // Show graph if the impact analysis is enable for $class
+      if (isset($CFG_GLPI['impact_asset_types'][$class])) {
+         self::loadLibs();
+         self::prepareImpactNetwork($item);
+         self::buildNetwork($item);
+      }
+
+      return true;
+   }
+
+   /**
+    * Load the cytoscape and spectrum-colorpicker librairies
+    *
+    * @since 9.5
+    */
+   public static function loadLibs() {
+      echo Html::css('public/lib/spectrum-colorpicker.css');
```

Comment:
There is another mechanism to load CSS and JS

Reply:
I think we did it that way because the files will be needed in a lot a pages (each asset type) and some plugin might add new assets type that will require them aswell. Can you confirm @cedric-anne ?